### PR TITLE
chore: create flaky test template

### PR DESCRIPTION
**Pull request by rachelruderman**
### User facing changelog
n/a

### Additional details
Adds an issue template for flaky internal tests. This is helpful for logging issues as part of the [proposal to Restore Confidence in CI](https://github.com/cypress-io/prod-eng-docs/blob/main/cypress-app/restore-confidence-in-ci.md)

### Steps to test
n/a

### How has the user experience changed?
n/a

### PR Tasks

- [na] Have tests been added/updated?
- [na] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/4-flaky-test.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
+name: "❄️ Flaky internal test"
+title: "[Flaky internal test]: "
+labels: ["topic: flake ❄️", "stage: fire watch"]
+epics: ["Restore confidence in CI"]
```
Comment (rachelruderman) [inline .github/ISSUE_TEMPLATE/4-flaky-test.yml:4]: TBH I don't know if this `epics` line will work, but there's no way to test it without merging. If it doesn't work, only the template will be broken -- nothing else -- so it's safe to try out

File: .github/ISSUE_TEMPLATE/4-flaky-test.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
+name: "❄️ Flaky internal test"
```
Comment (BlueWinds) [inline .github/ISSUE_TEMPLATE/4-flaky-test.yml:1]: This is going to be public, right? Intended for our use - but on the Cypress repo, so users will be able to see it and create issues using it.

Perhaps use a title like "Flaky test in `cypress-io/cypress` repository"? "Internal" may not be descriptive enough for people who don't already know what it means.

**Review by BlueWinds — APPROVED** [review]

**Review by marktnoonan — APPROVED** [review]
This is a good idea, and the tags will make it easy to check if a certain flaky test has been reported already.

File: .github/ISSUE_TEMPLATE/4-flaky-test.yml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+description: Report a flaky test in the Cypress app open-source code.
+body:
+  - type: markdown
+    attributes:
+      value: |
+        Have a question? 👉 [Start a new discussion](https://github.com/cypress-io/cypress/discussions) or [ask in chat](https://on.cypress.io/discord).
+  - type: textarea
+    id: dashboard
+    attributes:
+      label: Link to dashboard
```
Comment (mjhenkes) [inline .github/ISSUE_TEMPLATE/4-flaky-test.yml:14]:
```suggestion
      label: Link to dashboard or circle ci failure
```

File: .github/ISSUE_TEMPLATE/4-flaky-test.yml [hunk 4]
```diff
@@ -0,0 +1,45 @@
+name: "❄️ Flaky internal test"
+title: "[Flaky internal test]: "
+labels: ["topic: flake ❄️", "stage: fire watch"]
+epics: ["Restore confidence in CI"]
+description: Report a flaky test in the Cypress app open-source code.
```
Comment (mjhenkes) [inline .github/ISSUE_TEMPLATE/4-flaky-test.yml:5]:
```suggestion
description: Report a flaky test in the Cypress open-source repository.
```

File: .github/ISSUE_TEMPLATE/4-flaky-test.yml [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+      description: If you can, provide a quick analysis of why this test is flaky. 
+      placeholder: ex. The test appears to be flaky because...
+    validations:
+      required: true
+  - type: input
+    id: version
+    attributes:
+      label: Cypress Version
+      description: Run `cypress version` to see your current version. If possible, please update Cypress to the latest version first.
+      placeholder: ex. 7.6.0
```
Comment (mjhenkes) [inline .github/ISSUE_TEMPLATE/4-flaky-test.yml:38]: we can update our version number :)

```suggestion
      placeholder: ex.  10.4.0
```

File: .github/ISSUE_TEMPLATE/4-flaky-test.yml [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+      label: Analysis
+      description: If you can, provide a quick analysis of why this test is flaky. 
+      placeholder: ex. The test appears to be flaky because...
+    validations:
+      required: true
+  - type: input
+    id: version
+    attributes:
+      label: Cypress Version
+      description: Run `cypress version` to see your current version. If possible, please update Cypress to the latest version first.
```
Comment (mjhenkes) [inline .github/ISSUE_TEMPLATE/4-flaky-test.yml:37]:
```suggestion
      description: Provide the version of Cypress where the flake is occuring.
```

File: .github/ISSUE_TEMPLATE/4-flaky-test.yml [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+body:
+  - type: markdown
+    attributes:
+      value: |
+        Have a question? 👉 [Start a new discussion](https://github.com/cypress-io/cypress/discussions) or [ask in chat](https://on.cypress.io/discord).
+  - type: textarea
+    id: dashboard
+    attributes:
+      label: Link to dashboard
+      description: Please include a link to the failure in the Cypress Dashboard
```
Comment (mjhenkes) [inline .github/ISSUE_TEMPLATE/4-flaky-test.yml:15]:
```suggestion
      description: Please include a link to the failure in the Cypress Dashboard or in Circle Ci
```

**Pull request merged**